src/blueprints/Do.Blueprints.Service.Application/Business/Default/DefaultBusinessFeature.cs

```diff
@@ -11,60 +11,133 @@ namespace Do.Business.Default;
 public class DefaultBusinessFeature(List<Assembly> _domainAssemblies)
     : IFeature<BusinessConfigurator>
 {
-    const BindingFlags _defaultMemberBindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly;
+    const BindingFlags _bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly;
 
     public void Configure(LayerConfigurator configurator)
     {
-        configurator.ConfigureDomainAssemblyCollection(assemblies =>
+        configurator.ConfigureDomainTypeCollection(types =>
         {
             foreach (var assembly in _domainAssemblies)
             {
-                assemblies.Add(assembly);
+                types.AddFromAssembly(assembly, except: type =>
+                    (type.IsSealed && type.IsAbstract) || // if type is static
+                    type.IsAssignableTo(typeof(Exception)) ||
+                    type.IsAssignableTo(typeof(Attribute)) ||
+                    type.IsAssignableTo(typeof(Delegate))
+                );
             }
         });
 
         configurator.ConfigureDomainBuilderOptions(options =>
         {
-            options.ConstuctorBindingFlags = _defaultMemberBindingFlags;
-            options.MethodBindingFlags = _defaultMemberBindingFlags;
-            options.PropertyBindingFlags = _defaultMemberBindingFlags;
+            options.BindingFlags.Constructor = _bindingFlags;
+            options.BindingFlags.Method = _bindingFlags;
+            options.BindingFlags.Property = _bindingFlags;
+
+            options.BuildLevels.Add(context => context.DomainTypesContain(context.Type), BuildLevels.Members);
+            options.BuildLevels.Add(context => context.Type.IsGenericType && context.DomainTypesContain(context.Type.GetGenericTypeDefinition()), BuildLevels.Members);
+            options.BuildLevels.Add(type => !type.IsValueType, BuildLevels.Inheritance);
+            options.BuildLevels.Add(type => type.IsGenericType, BuildLevels.Generics);
+        });
+
+        configurator.ConfigureDomainIndexOptions(options =>
+        {
+            options.Type.Add(AttributeIndex.New<ServiceAttribute>());
+            options.Type.Add(AttributeIndex.New<TransientAttribute>());
+            options.Type.Add(AttributeIndex.New<ScopedAttribute>());
+            options.Type.Add(AttributeIndex.New(new SingletonAttribute()));
```

Comment:
accepting an instance is not a good way to get type. here we need the type not the instance, so `New(typeof(SingletonAttribute))` should be preferred.

combined with previous comment, it is better to have `options.Type.Add<SingletonAttribute>()` and `options.Type.Add(typeof(SingletonAttribute))` options in case you have the type during runtime.